Unprefix font-size: -webkit-xxx-large

nt1m opened the pull request:
#### 07860fac0bbf23c58f3e2519f46cc8de480452c8
<pre>
Unprefix font-size: -webkit-xxx-large
<a href="https://bugs.webkit.org/show_bug.cgi?id=246574">https://bugs.webkit.org/show_bug.cgi?id=246574</a>
rdar://101207950

Reviewed by Myles C. Maxfield.

xxx-large is now part of the spec and matches what we&apos;ve implemented: <a href="https://drafts.csswg.org/css-fonts-4/#valdef-font-size-absolute-size">https://drafts.csswg.org/css-fonts-4/#valdef-font-size-absolute-size</a>

Make -webkit-xxx-large an alias of xxx-large at parse-time, so it&apos;s easy to remove later on.

* LayoutTests/editing/style/push-down-font-styles-mac-expected.txt:
* LayoutTests/editing/style/push-down-font-styles-mac.html:
* LayoutTests/editing/style/push-down-font-styles-win-expected.txt:
* LayoutTests/editing/style/push-down-font-styles-win.html:
* LayoutTests/fast/css/font-size-webkit-xxx-large-expected.html: Added.
* LayoutTests/fast/css/font-size-webkit-xxx-large.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/font-size-xxx-large.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-valid-expected.txt:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::consumeFontSize):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeFontSizeRaw):
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::legacyFontSizeFromCSSValue):
* Source/WebCore/html/HTMLFontElement.cpp:
(WebCore::HTMLFontElement::cssValueFromFontSizeNumber):
* Source/WebCore/platform/graphics/FontCascadeDescription.h:
(WebCore::FontCascadeDescription::keywordSizeAsIdentifier const):
(WebCore::FontCascadeDescription::setKeywordSizeFromIdentifier):
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::BuilderCustom::applyValueFontSize):
* Source/WebCore/style/StyleFontSizeFunctions.h:
* Source/WebCore/style/StyleResolveForFontRaw.cpp:
(WebCore::Style::resolveForFontRaw):
* Source/WebInspectorUI/UserInterface/Models/CSSKeywordCompletions.js:

Canonical link: <a href="https://commits.webkit.org/255602@main">https://commits.webkit.org/255602@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5f6ad182f9444c5d47e11d9839c42cdefd8a3d46

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93054 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2255 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23650 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102763 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/162988 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/97056 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2265 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30583 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/85437 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98879 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98719 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/1558 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79528 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/28474 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/83472 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83199 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/71586 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36977 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/17102 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34789 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18315 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3878 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38661 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40894 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40588 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/37508 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

litherum commented on LayoutTests/editing/style/push-down-font-styles-mac.html:
```diff
@@ -66,9 +66,9 @@
 testSingleToggle("fontsize", 2, selectAll, 'hello world', '<font size="2">hello world</font>');
 testSingleToggle("fontsize", 4, selectAll, 'hello world', '<font size="4">hello world</font>');
 testSingleToggle("fontsize", 5, selectFirstWord, 'hello world', '<font size="5">hello</font> world');
-testSingleToggle("fontsize", 3, selectFirstWord, '<font size="7">hello <div>world</div></font>', 'hello <div style="font-size: -webkit-xxx-large;">world</div>');
```
After this patch, will we have any tests that make sure the prefixed version still works?

litherum replied:
Oh, yes, I see it below.

darinadler reviewed (approved):
On similar changes in the past I have seen people file a bug about removing the WebKit-prefixed version, and mention it in all the places the code will eventually be removed.

I’m not sure if that’s best practice or just clutter, though.

nt1m committed:
Unprefix font-size: -webkit-xxx-large
https://bugs.webkit.org/show_bug.cgi?id=246574
rdar://101207950

Reviewed by Myles C. Maxfield.

xxx-large is now part of the spec and matches what we've implemented: https://drafts.csswg.org/css-fonts-4/#valdef-font-size-absolute-size

Make -webkit-xxx-large an alias of xxx-large at parse-time, so it's easy to remove later on.

* LayoutTests/editing/style/push-down-font-styles-mac-expected.txt:
* LayoutTests/editing/style/push-down-font-styles-mac.html:
* LayoutTests/editing/style/push-down-font-styles-win-expected.txt:
* LayoutTests/editing/style/push-down-font-styles-win.html:
* LayoutTests/fast/css/font-size-webkit-xxx-large-expected.html: Added.
* LayoutTests/fast/css/font-size-webkit-xxx-large.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/font-size-xxx-large.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-valid-expected.txt:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::consumeFontSize):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeFontSizeRaw):
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::legacyFontSizeFromCSSValue):
* Source/WebCore/html/HTMLFontElement.cpp:
(WebCore::HTMLFontElement::cssValueFromFontSizeNumber):
* Source/WebCore/platform/graphics/FontCascadeDescription.h:
(WebCore::FontCascadeDescription::keywordSizeAsIdentifier const):
(WebCore::FontCascadeDescription::setKeywordSizeFromIdentifier):
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::BuilderCustom::applyValueFontSize):
* Source/WebCore/style/StyleFontSizeFunctions.h:
* Source/WebCore/style/StyleResolveForFontRaw.cpp:
(WebCore::Style::resolveForFontRaw):
* Source/WebInspectorUI/UserInterface/Models/CSSKeywordCompletions.js:

Canonical link: https://commits.webkit.org/255602@main
